flutter_keyboard_visibility/lib/src/ui/keyboard_dismiss_on_tap.dart

```diff
@@ -7,27 +7,69 @@ import 'package:flutter/widgets.dart';
 /// tree and when the user taps outside of a focused widget,
 /// the focus will be removed and the keyboard will be hidden.
 class KeyboardDismissOnTap extends StatelessWidget {
+  /// Used to control if clicking a widget inheriting GestureDetector will
+  /// dismiss the keyboard or not
+  ///
+  /// By default GestureDetectors are ignored
+  final bool ignoreGestureDetectors;
+
+  /// This is used to control if the next dismiss is allowed
+  static bool _dismissible = true;
```

Comment:
Since this is static, it will lead to inconsistent results with multiple widget trees using KeyboardDismissOnTap at the same time.

The proper way to modify state up the tree is to use an InheritedWidget. Docs here: https://api.flutter.dev/flutter/widgets/InheritedWidget-class.html

We use this pattern in this repo today, if you take a look at KeyboardVisibilityProvider.